### PR TITLE
add create_service_endpoint activity

### PR DESCRIPTION
**Pull request by tdevilleduc**
Signed-off-by: Thomas Deville-Duc <devilleduc.thomas@gmail.com>

### CODE REVIEW

File: chaosk8s/actions.py [hunk 1]
```diff
@@ -143,3 +143,21 @@ def scale_microservice(name: str, replicas: int, ns: str = "default",
         raise ActivityFailed(
             "failed to scale '{s}' to {r} replicas: {e}".format(
                 s=name, r=replicas, e=str(e)))
+
+
+def scale_statefulset(name: str, replicas: int, ns: str = "default",
```
Comment (dmartin35) [inline chaosk8s/actions.py]: this activity is not exported in the `__all__` list

**Comment by dmartin35** [comment]
thanks @tdevilleduc for the PR
could you please fixup/squash your 'fix code syntax' commits to each appropriate activity commits. To have only two proper commits (one per activity)
thanks

**Comment by Lawouach** [comment]
Hi   @tdevilleduc, do you mind rebasing from master please? (rebase not merge commit please)

**Comment by Lawouach** [comment]
It's not very typical of this extension to read resources from local files. I need to ponder on this.

**Comment by tdevilleduc** [comment]
> Hi @tdevilleduc, do you mind rebasing from master please? (rebase not merge commit please)

I don't understand what you are asking me.  I tried something

**Comment by tdevilleduc** [comment]
Ok ! Wasn't easy, but i think it's ok now :)

**Comment by Lawouach** [comment]
Thanks! This helps a lot the review/merge indeed :)

**Comment by tdevilleduc** [comment]
> It's not very typical of this extension to read resources from local files. I need to ponder on this.

I copied the behavior seen in other places :) 
The other way is to recreate file structure with data. Its possible for service structure, but it can be really tough for microservice (the other place i copied it from)

**Comment by Lawouach** [comment]
> I copied the behavior seen in other places :)

Hah! Good catch. I had forgotten about it and I should have been firmer on my review. 

> The other way is to recreate file structure with data. Its possible for service structure, but it can be really tough for microservice

I agree. This wouldn't be as simple as having these structures in files.

Good enough.

**Comment by Lawouach** [comment]
Hello,

I am sorry I have taken so long to come back to this but the end of year has been way too busy unfortunately.

So, I'm looking at this PR again. I think the create_service_endpoint should be part of its own subpackage. You added it to `chaosk8s/actions.py` but it should be `chaosk8s/service/actions.py`

The `chaosk8s/actions.py` should not really be used anymore, it's a legacy module that dates from the early days of this extension.

Also, you copied the entire block of code from another action indeed but I'd rather avoid duplicated code, it would be better to have a function that knows how to load a k8s manifest. This should live in `chaosk8s/__init__.py` and be imported wherever you want to apply a manifest.

File: chaosk8s/actions.py [hunk 2]
```diff
@@ -143,3 +144,21 @@ def scale_microservice(name: str, replicas: int, ns: str = "default",
...
```
Comment (Lawouach) [inline chaosk8s/actions.py]: should this belong to a different PR since this PR is about create_service?

Comment (tdevilleduc) [reply chaosk8s/actions.py]: I will wait for https://github.com/chaostoolkit/chaostoolkit-kubernetes/pull/70 approval before creating a new PR. I've made several updates from the same master branch, its a bit confusing
Nevertheless, you're right. I will remove this part from this PR

File: chaosk8s/actions.py [hunk 3]
```diff
@@ -95,6 +95,28 @@ def kill_microservice(name: str, ns: str = "default",
             p.metadata.name, ns, body=body)
 
 
+def create_service_endpoint(spec_path: str, ns: str = "default",
```
[the remark this answers is not in the document]
Comment (tdevilleduc) [reply chaosk8s/actions.py]: It's done.
I didn't move remove_service_endpoint yet, i will do it in an other PR after this one is approved

File: chaosk8s/actions.py (hunk 2 above)
Comment (tdevilleduc) [reply chaosk8s/actions.py]: https://github.com/chaostoolkit/chaostoolkit-kubernetes/pull/73

**Comment by Lawouach** [comment]
can we kill this one now @tdevilleduc ?

**Comment by tdevilleduc** [comment]
> can we kill this one now @tdevilleduc ?

This one is also useful : 
I have an experiment where i remove a service_endpoint (with remove_service_endpoint) and i want to recreate it at the end of the experiment
If you're ok with it, I have already rework it to use a service module, and I will add some test + changelog

**Comment by tdevilleduc** [comment]
Done

File: chaosk8s/service/actions.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+import json
+import os.path
+
+import yaml
```
Comment (Lawouach) [inline chaosk8s/service/actions.py]: This one import must be the last of this block :)
